### PR TITLE
Mds/feature/7 create unit tests for core classes

### LINKED ISSUE #7: create unit tests for core classes
Create unit tests for:
Shop
Product
User
Merchant
Buyer
Cart
ItemQuantityList
Main

### PR DESCRIPTION
**Pull request by mpdesantis**
## PR type

- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Performance Optimization
- [ ] Documentation Update

## Description
This PR is motivated by the need to provide unit testing for core classes.

It introduces the following changes:
* Adds JUnit Test Suite for `Buyer` class
* Adds JUnit Test Suite for `Cart` class
* Adds JUnit Test Suite for `ItemQuantityList` class
* Adds JUnit Test Suite for `Merchant` class
* Adds JUnit Test Suite for `Product` class
* Adds JUnit Test Suite for `Shop` class
* Adds JUnit Test Suite for `User` class
* Fixes minor bug in `Cart` class
* Fixes minor bug in `ItemQuantityList` class
* Adds values for `Tag` enum to facilitate testing
* Adds method to `Product` class to facilitate `Tag` testing
* Updates ignores
## Related Issues/Tickets

- Closes #7 
## QA Instruction, Screenshots etc...

Run all JUnit test classes found in `src/test/java/SYSC4806Project/`.

## Added/updated tests?

- [x] Yes
- [ ] No: _replace this line with details on why tests are not included in this PR_

## PR Checklist

- [x] I have rebased this branch against develop/latest
- [ ] All the tests are passing
__N.B.__ Not all tests are passing, as these unit test suites test methods not fully implemented, and test failures due to bugs are already logged as GitHub issues (#30, #29, #10). 
- [x] My commit messages are descriptive and useful

### CODE REVIEW

**Review by ArthurAtangana — APPROVED** [review]

File: src/main/java/SYSC4806Project/Cart.java
Comment (mpdesantis) [inline src/main/java/SYSC4806Project/Cart.java:1]: @Rebecca-Elliott Noting this change, as I slightly changed your logic. I think the `removeItem()` should wrap `removeItems()` and not `removeProduct()` - I'm assuming it was a copy/paste typo, but can you check that it makes sense? Thanks!

File: src/main/java/SYSC4806Project/ItemQuantityList.java [hunk 1]
```diff
@@ -40,7 +43,7 @@ public boolean addItems(Product product, int quantity) {
      * @return true if the product was added
      */
     public boolean addProduct(Product product) {
-        if (product == null || !this.contains(product)) {return false;}
```
Comment (mpdesantis) [inline src/main/java/SYSC4806Project/ItemQuantityList.java:43]: @Rebecca-Elliott From testing, I think that was just a flipped condition - removed the `!`. I think the check should be to see if the `Product` is `null` or is already in the `ItemQuantityList`. Can you check this makes sense? Thanks!

File: src/main/java/SYSC4806Project/ItemQuantityList.java [hunk 2]
```diff
@@ -57,7 +60,7 @@ public boolean removeItems(Product product, int quantity) {
                     System.out.println("Attempted to remove more quantity than available. None were removed.");
                     return false;
                 }
-                itemQuantityPair.setQuantity(quantity - itemQuantityPair.getQuantity());
+                itemQuantityPair.setQuantity(itemQuantityPair.getQuantity() - quantity);
```
Comment (mpdesantis) [inline src/main/java/SYSC4806Project/ItemQuantityList.java:63]: @Rebecca-Elliott Just switched the subtraction order to "existing quantity - quantity to remove". Can you check this makes sense? Thanks!

File: src/main/java/SYSC4806Project/ItemQuantityList.java [hunk 3]
```diff
@@ -101,7 +104,7 @@ public int getItemQuantity(Product product) {
     }
 
 
-    public class ItemQuantityPair {
+    public static class ItemQuantityPair {
```
Comment (mpdesantis) [inline src/main/java/SYSC4806Project/ItemQuantityList.java:107]: IntelliJ forced me to make this static when I was writing the test classes. Any objections?

File: src/main/java/SYSC4806Project/Cart.java
Comment (Rebecca-Elliott) [reply src/main/java/SYSC4806Project/Cart.java:1]: correct, thank you.

File: src/main/java/SYSC4806Project/ItemQuantityList.java (hunk 1 above)
Comment (Rebecca-Elliott) [reply src/main/java/SYSC4806Project/ItemQuantityList.java:43]: yes it was a typo.

File: src/main/java/SYSC4806Project/ItemQuantityList.java (hunk 2 above)
Comment (Rebecca-Elliott) [reply src/main/java/SYSC4806Project/ItemQuantityList.java:63]: correct.

File: src/main/java/SYSC4806Project/ItemQuantityList.java (hunk 3 above)
Comment (Rebecca-Elliott) [reply src/main/java/SYSC4806Project/ItemQuantityList.java:126]: I don't think there is any problems doing it this way.

**Pull request merged**